models/idea/index.js

```diff
@@ -216,6 +216,63 @@ class Idea extends Model {
     const cursor = await this.db.query(query, params);
     return await cursor.all();
   }
+
+  /**
+   * Read ideas with specified creators
+   * @param {string[]} usernames - list of usernames to search with
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findWithCreators(creators, { offset, limit }) {
+    // TODO  to be checked for query optimization or unnecessary things
+    const query = `
+      FOR idea IN ideas
+        // sort from newest
+        SORT idea.created DESC
+        LIMIT @offset, @limit
+        // find creator
+        LET c = (DOCUMENT(idea.creator))
+        LET creator = MERGE(KEEP(c, 'username'), c.profile)
+        // format for output
+        LET ideaOut = MERGE(KEEP(idea, 'title', 'detail', 'created'), { id: idea._key}, { creator })
+        // limit
+        // respond
+        FILTER creator.username in @creators
+        RETURN ideaOut`;
+
+    const params = { offset, limit , creators };
+    const cursor = await this.db.query(query, params);
+    return await cursor.all();
+  }
+
+
+  /**
+   * Read ideas commented by specified users
+   * @param {string[]} usernames - list of usernames to search with
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findCommentedBy(commentedBy, { offset, limit }) {
+
+    const query = `
+      FOR user IN users 
+      FILTER user.username IN @commentedBy
+        FOR comment IN comments 
+        FILTER comment.creator == user._id 
+        AND SUBSTRING(comment.primary, 0, 5) == 'ideas' 
```

Comment:
This should be replaced by more appropriate [`IS_SAME_COLLECTION`](https://docs.arangodb.com/3.3/AQL/Functions/Document.html#issamecollection]) or [`PARSE_IDENTIFIER`](https://docs.arangodb.com/3.3/AQL/Functions/Document.html#parseidentifier) functions.